### PR TITLE
Use singular names for time units, as in Erlang/OTP 19.1. Fix #5236

### LINKED ISSUE #5236: Deprecate seconds, milliseconds, microseconds and nanoseconds
Erlang 19.1 standardizes on the singular forms, so we shall do the same.

### PR DESCRIPTION
**Pull request by alco**

### CODE REVIEW

File: lib/elixir/lib/system.ex [hunk 1]
```diff
@@ -53,7 +53,7 @@ defmodule System do
   the time, all calculations are done in the `:native` unit, to
   avoid loss of precision, with `convert_time_unit/3` being
   invoked at the end to convert to a specific time unit like
-  milliseconds or microseconds. See the `t:time_unit/0` type for
+  `millisecond` or `microsecond`. See the `t:time_unit/0` type for
```
Comment (whatyouhide) [inline lib/elixir/lib/system.ex]: If we're using backticks here I guess we should use the full atom, no? Like `:millisecond` or `:microsecond`.

File: lib/elixir/lib/system.ex [hunk 2]
```diff
@@ -65,25 +65,31 @@ defmodule System do
...
+  parts_per_second` seconds. For example, using the `:millisecond` time unit
   is equivalent to using `1000` as the time unit (as the time will be returned
   in 1/1000 seconds - milliseconds).
 
-  Keep in mind the Erlang API will use `:milli_seconds`, `:micro_seconds`
-  and `:nano_seconds` as time units although Elixir normalizes their spelling
-  to match the SI convention.
+  Keep in mind the Erlang API prior to version 19.1 will use `:milli_seconds`,
+  `:micro_seconds` and `:nano_seconds` as time units although Elixir normalizes
+  their spelling to match the SI convention.
```
Comment (whatyouhide) [inline lib/elixir/lib/system.ex]: If we mention that Erlang < 19.1 will use `:milli_seconds` and such maybe we should mention that Erlang >= 19.1 will use `:milli_second` and such (still with the underscore)?

File: lib/elixir/lib/system.ex [hunk 3]
```diff
@@ -65,25 +65,31 @@ defmodule System do
...
+    | :microsecond
+    | :nanosecond
+    | pos_integer
+    # TODO: Deprecate these in Elixir 2.0
+    | :seconds
     | :milliseconds
     | :microseconds
     | :nanoseconds
-    | pos_integer
+
```
Comment (whatyouhide) [inline lib/elixir/lib/system.ex]: ✂️

File: lib/elixir/lib/system.ex (hunk 1 above)
Comment (alco) [inline lib/elixir/lib/system.ex]: Definitely! I'll update it.

File: lib/elixir/lib/system.ex (hunk 2 above)
Comment (alco) [inline lib/elixir/lib/system.ex]: Erlang 19.1 uses the same names as Elixir in this PR, search for "millisecond" here – http://erlang.org/download/otp_src_19.1.readme. So if at some point Elixir drops support for earlier version of Erlang, this paragraph and the conversions can be removed completely.

File: lib/elixir/lib/system.ex (hunk 3 above)
Comment (alco) [inline lib/elixir/lib/system.ex]: I've removed the empty line.

File: lib/elixir/lib/system.ex (hunk 1 above)
Comment (alco) [inline lib/elixir/lib/system.ex]: Updated.

File: lib/elixir/lib/system.ex (hunk 2 above)
Comment (whatyouhide) [inline lib/elixir/lib/system.ex]: Oh, I grepped the page for `second` and glanced the underscore and assumed they just fixed the singular/plural, not the underscore 😄 Awesome then!

**Pull request merged**